### PR TITLE
transition from paperspace to gradient

### PR DESCRIPTION
**Pull request by freezeboy**
###### Motivation for this change

This PR contains some changes to resolve the reverse dependencies broken by the automatic PR #103814.

Upstream paperspace has worked to rewrite its SDK which involves an update of the complete stack. The dangerous this here is this update includes `wheel` and `numpy` (even though for this one, it is a revision update).

If you want to run nixpkgs-review, prepare for a big build I guess, I compiled the packages inside this PR only.

This will not compile before PR #103814 is merged.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by freezeboy** [comment]
Now hyperopt has been updated, this PR should be building correctly

**Review by thoughtpolice — CHANGES_REQUESTED** [review]
Thanks, but you'll definitely need to retarget this to be based on the `staging` branch instead, because it will effectively rebuild the world. We can't merge this until then, so I'm just putting in a soft `Request Changes` &mdash; other than that your changes look fine to me, but someone like Jon or Frederik should take a look.

Note that GitHub handles moving PRs onto new base branches very poorly (so poorly, in fact, that I cannot even describe how to do it properly; everybody messes it up completely for a reason, I suppose!) and it often ends up pinging hundreds of maintainers in the process, so you might actually want to just close this PR and open a new one...

**Comment by freezeboy** [comment]
Wouldn't it be enough to rebase on staging on my side then push force here?

**Comment by freezeboy** [comment]
Ok rebased on staging, looks good to me in terms of PR content

It seems numpy world-rebuild was already part of staging, so it vanished for this PR

**Comment by SuperSandro2000** [comment]
Can you solve the merge conflict?

**Comment by jonringer** [comment]
is the wheel bump necessary? it's causing the entire python package set to be rebuilt

**Comment by freezeboy** [comment]
It didn't try to remove the strict dependency for it, so not sure, I just applied the requirement of gradient tool. But it is true this is quite a big change :/

File: pkgs/development/python-modules/marshmallow/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/marshmallow/default.nix:27]:
```suggestion
    description = "lightweight library for converting complex objects to and from simple Python datatypes";
```

**Comment by freezeboy** [comment]
Changed the description and squashed previous fixups

**Comment by SuperSandro2000** [comment]
Please target master cause this is not a mass rebuild.

File: pkgs/development/python-modules/gradient-utils/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/gradient-utils/default.nix:20]:
```suggestion
```

File: pkgs/development/python-modules/gradient-utils/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/gradient-utils/default.nix:17]:
```suggestion
      --replace "numpy = \"1.19.3\"" "numpy"
```

**Comment by freezeboy** [comment]
> Please target master cause this is not a mass rebuild.

It is no longer the case as wheel as been bumped by another PR I guess

**Comment by rmcgibbo** [comment]
Result of `nixpkgs-review pr 103841` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages failed to build:</summary>
  <ul>
    <li>apache-airflow (python37Packages.apache-airflow)</li>
    <li>google-music-scripts</li>
    <li>python37Packages.dftfit</li>
    <li>python37Packages.google-music</li>
    <li>python37Packages.google-music-proto</li>
    <li>python37Packages.hug</li>
    <li>python38Packages.dftfit</li>
    <li>python38Packages.google-music</li>
    <li>python38Packages.google-music-proto</li>
    <li>python38Packages.hug</li>
    <li>python39Packages.dataclasses-json</li>
    <li>python39Packages.google-music-proto</li>
    <li>python39Packages.hug</li>
  </ul>
</details>
<details>
  <summary>45 packages built:</summary>
  <ul>
    <li>esphome</li>
    <li>flashfocus</li>
    <li>platformio</li>
    <li>python37Packages.apispec</li>
    <li>python37Packages.dataclasses-json</li>
    <li>python37Packages.flask-appbuilder</li>
    <li>python37Packages.flask_marshmallow</li>
    <li>python37Packages.gradient</li>
    <li>python37Packages.gradient-utils</li>
    <li>python37Packages.halo</li>
    <li>python37Packages.log-symbols</li>
    <li>python37Packages.marshmallow</li>
    <li>python37Packages.marshmallow-enum</li>
    <li>python37Packages.marshmallow-polyfield</li>
    <li>python37Packages.marshmallow-sqlalchemy</li>
    <li>python37Packages.marshmallow2</li>
    <li>python37Packages.spinners</li>
    <li>python38Packages.apispec</li>
    <li>python38Packages.dataclasses-json</li>
    <li>python38Packages.flask-appbuilder</li>
    <li>python38Packages.flask_marshmallow</li>
    <li>python38Packages.gradient</li>
    <li>python38Packages.gradient-utils</li>
    <li>python38Packages.halo</li>
    <li>python38Packages.log-symbols</li>
    <li>python38Packages.marshmallow</li>
    <li>python38Packages.marshmallow-enum</li>
    <li>python38Packages.marshmallow-polyfield</li>
    <li>python38Packages.marshmallow-sqlalchemy</li>
    <li>python38Packages.marshmallow2</li>
    <li>python38Packages.spinners</li>
    <li>python39Packages.apispec</li>
    <li>python39Packages.flask-appbuilder</li>
    <li>python39Packages.flask_marshmallow</li>
    <li>python39Packages.gradient</li>
    <li>python39Packages.gradient-utils</li>
    <li>python39Packages.halo</li>
    <li>python39Packages.log-symbols</li>
    <li>python39Packages.marshmallow</li>
    <li>python39Packages.marshmallow-enum</li>
    <li>python39Packages.marshmallow-polyfield</li>
    <li>python39Packages.marshmallow-sqlalchemy</li>
    <li>python39Packages.marshmallow2</li>
    <li>python39Packages.spinners</li>
    <li>sublime-music</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Please either execute tests for those packages or add ``doCheck = false;`` and pythonImportsCheck.

```
python38Packages.marshmallow.log                                                                                                                                                            190:Ran 0 tests in 0.000s                                                                                                                                                                                                                                                                                                                                                               sublime-music.log                                                                                                                                                                           487:Ran 0 tests in 0.000s                                                                                                                                                                                                                                                                                                                                                               python37Packages.marshmallow2.log                                                                                                                                                           206:Ran 0 tests in 0.000s                                                                                                                                                                                                                                                                                                                                                               python39Packages.marshmallow2.log                                                                                                                                                           206:Ran 0 tests in 0.000s                                                                                                                                                                                                                                                                                                                                                               python39Packages.gradient-utils.log                                                                                                                                                         153:Ran 0 tests in 0.000s                                                                                                                                                                                                                                                                                                                                                               python39Packages.marshmallow.log                                                                                                                                                            190:Ran 0 tests in 0.000s                                                                                                                                                                                                                                                                                                                                                               python38Packages.gradient-utils.log                                                                                                                                                         153:Ran 0 tests in 0.000s                                                                                                                                                                                                                                                                                                                                                               python38Packages.dataclasses-json.log                                                                                                                                                       159:Ran 0 tests in 0.000s                                                                                                                                                                                                                                                                                                                                                               python37Packages.marshmallow.log                                                                                                                                                            190:Ran 0 tests in 0.000s                                                                                                                                                                                                                                                                                                                                                               python37Packages.dataclasses-json.log                                                                                                                                                       159:Ran 0 tests in 0.000s                                                                                                                                                                                                                                                                                                                                                               python38Packages.marshmallow2.log                                                                                                                                                           206:Ran 0 tests in 0.000s                                                                                                                                                                                                                                                                                                                                                               python37Packages.gradient-utils.log                                                                                                                                                         153:Ran 0 tests in 0.000s
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103841` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li>google-music-scripts</li>
    <li>python37Packages.dftfit</li>
    <li>python37Packages.google-music</li>
    <li>python37Packages.google-music-proto</li>
    <li>python37Packages.hug</li>
    <li>python38Packages.dftfit</li>
    <li>python38Packages.google-music</li>
    <li>python38Packages.google-music-proto</li>
    <li>python38Packages.hug</li>
    <li>python39Packages.dataclasses-json</li>
    <li>python39Packages.google-music-proto</li>
    <li>python39Packages.hug</li>
  </ul>
</details>
<details>
  <summary>46 packages built:</summary>
  <ul>
    <li>apache-airflow (python37Packages.apache-airflow)</li>
    <li>esphome</li>
    <li>flashfocus</li>
    <li>platformio</li>
    <li>python37Packages.apispec</li>
    <li>python37Packages.dataclasses-json</li>
    <li>python37Packages.flask-appbuilder</li>
    <li>python37Packages.flask_marshmallow</li>
    <li>python37Packages.gradient</li>
    <li>python37Packages.gradient-utils</li>
    <li>python37Packages.halo</li>
    <li>python37Packages.log-symbols</li>
    <li>python37Packages.marshmallow</li>
    <li>python37Packages.marshmallow-enum</li>
    <li>python37Packages.marshmallow-polyfield</li>
    <li>python37Packages.marshmallow-sqlalchemy</li>
    <li>python37Packages.marshmallow2</li>
    <li>python37Packages.spinners</li>
    <li>python38Packages.apispec</li>
    <li>python38Packages.dataclasses-json</li>
    <li>python38Packages.flask-appbuilder</li>
    <li>python38Packages.flask_marshmallow</li>
    <li>python38Packages.gradient</li>
    <li>python38Packages.gradient-utils</li>
    <li>python38Packages.halo</li>
    <li>python38Packages.log-symbols</li>
    <li>python38Packages.marshmallow</li>
    <li>python38Packages.marshmallow-enum</li>
    <li>python38Packages.marshmallow-polyfield</li>
    <li>python38Packages.marshmallow-sqlalchemy</li>
    <li>python38Packages.marshmallow2</li>
    <li>python38Packages.spinners</li>
    <li>python39Packages.apispec</li>
    <li>python39Packages.flask-appbuilder</li>
    <li>python39Packages.flask_marshmallow</li>
    <li>python39Packages.gradient</li>
    <li>python39Packages.gradient-utils</li>
    <li>python39Packages.halo</li>
    <li>python39Packages.log-symbols</li>
    <li>python39Packages.marshmallow</li>
    <li>python39Packages.marshmallow-enum</li>
    <li>python39Packages.marshmallow-polyfield</li>
    <li>python39Packages.marshmallow-sqlalchemy</li>
    <li>python39Packages.marshmallow2</li>
    <li>python39Packages.spinners</li>
    <li>sublime-music</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103841` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python37Packages.dftfit</li>
    <li>python38Packages.dftfit</li>
    <li>python39Packages.google-music</li>
    <li>sublime-music</li>
  </ul>
</details>
<details>
  <summary>11 packages failed to build:</summary>
  <ul>
    <li>apache-airflow (python37Packages.apache-airflow)</li>
    <li>google-music-scripts</li>
    <li>python37Packages.google-music</li>
    <li>python37Packages.google-music-proto</li>
    <li>python37Packages.hug</li>
    <li>python38Packages.google-music</li>
    <li>python38Packages.google-music-proto</li>
    <li>python38Packages.hug</li>
    <li>python39Packages.dataclasses-json</li>
    <li>python39Packages.google-music-proto</li>
    <li>python39Packages.hug</li>
  </ul>
</details>
<details>
  <summary>41 packages built:</summary>
  <ul>
    <li>python37Packages.apispec</li>
    <li>python37Packages.dataclasses-json</li>
    <li>python37Packages.flask-appbuilder</li>
    <li>python37Packages.flask_marshmallow</li>
    <li>python37Packages.gradient</li>
    <li>python37Packages.gradient-utils</li>
    <li>python37Packages.halo</li>
    <li>python37Packages.log-symbols</li>
    <li>python37Packages.marshmallow</li>
    <li>python37Packages.marshmallow-enum</li>
    <li>python37Packages.marshmallow-polyfield</li>
    <li>python37Packages.marshmallow-sqlalchemy</li>
    <li>python37Packages.marshmallow2</li>
    <li>python37Packages.spinners</li>
    <li>python38Packages.apispec</li>
    <li>python38Packages.dataclasses-json</li>
    <li>python38Packages.flask-appbuilder</li>
    <li>python38Packages.flask_marshmallow</li>
    <li>python38Packages.gradient</li>
    <li>python38Packages.gradient-utils</li>
    <li>python38Packages.halo</li>
    <li>python38Packages.log-symbols</li>
    <li>python38Packages.marshmallow</li>
    <li>python38Packages.marshmallow-enum</li>
    <li>python38Packages.marshmallow-polyfield</li>
    <li>python38Packages.marshmallow-sqlalchemy</li>
    <li>python38Packages.marshmallow2</li>
    <li>python38Packages.spinners</li>
    <li>python39Packages.apispec</li>
    <li>python39Packages.flask-appbuilder</li>
    <li>python39Packages.flask_marshmallow</li>
    <li>python39Packages.gradient</li>
    <li>python39Packages.gradient-utils</li>
    <li>python39Packages.halo</li>
    <li>python39Packages.log-symbols</li>
    <li>python39Packages.marshmallow</li>
    <li>python39Packages.marshmallow-enum</li>
    <li>python39Packages.marshmallow-polyfield</li>
    <li>python39Packages.marshmallow-sqlalchemy</li>
    <li>python39Packages.marshmallow2</li>
    <li>python39Packages.spinners</li>
  </ul>
</details>

**Pull request closed without merging**
